src/actions/heap/heap.ts

```diff
@@ -296,19 +303,24 @@ export class HeapAction extends Hub.Action {
     envId: string,
     requestUrl: string,
     heapField: HeapField,
+    errors: Error[],
   ): Promise<void> {
     const requestBody = this.constructBodyForRequest(
       envId,
       heapField,
       requestBatch,
     )
-    await req
-      .post({
-        uri: requestUrl,
-        headers: { "Content-Type": "application/json" },
-        body: requestBody,
-      })
-      .promise()
+    try {
```

Comment:
Might this try statement be cover what the previous one is catching. It seemed to me like 2 try statements are covering same things.
Personally I would rather write this with promise.catch to avoid try/catch. But this version is also OK.

Reply:
Not sure which `try` block you're referring to by the "previous one." Do you mean the one in the `execute` method? I think the issue is that this promise can throws before the `await Promise.all(...)` line is executed in the `execute` method.